### ISSUE TITLE
Error on `require 'slop'` with Ruby 2.0.0

### ISSUE THREAD

**Issue by bjoernalbers** [leejarvis/slop#207]
Hello Lee,

when I load slop with Ruby 2.0 I get this error:

```
$ ruby -v
ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
$ ruby -r slop
/Library/Ruby/Gems/2.0.0/gems/slop-4.4.2/lib/slop/parser.rb:148:in `private': nil is not a symbol (TypeError)
        from /Library/Ruby/Gems/2.0.0/gems/slop-4.4.2/lib/slop/parser.rb:148:in `<class:Parser>'
        from /Library/Ruby/Gems/2.0.0/gems/slop-4.4.2/lib/slop/parser.rb:2:in `<module:Slop>'
        from /Library/Ruby/Gems/2.0.0/gems/slop-4.4.2/lib/slop/parser.rb:1:in `<top (required)>'
        from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /Library/Ruby/Gems/2.0.0/gems/slop-4.4.2/lib/slop.rb:3:in `<top (required)>'
        from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:133:in `require'
        from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:133:in `rescue in require'
        from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:40:in `require'
```

Ruby 2.1.3 works fine:

```
$ ruby -v
ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-darwin13.0]
$ ruby -r slop
^Cruby: Interrupt
```

It tested both Rubies with Slop 4.4.2 on mac OS Sierra:

```
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.12.4
BuildVersion:   16E195
```

It looks like this the "private" suffix (line 148 in `lib/parser.rb`) is only the problem:

```ruby
    private def partition(strings)
```

I wonder if this is necessary since `private` is already declared some lines above.

It would be cool if slop would support Ruby 2.0.0 since this is still the default Ruby on the latest macOS version.
But If you plan to drop Ruby 2.0-support you might want to update the README :-)

Björn

**Comment by leejarvis** [comment]
Hi Björn,

Thanks for reporting! That's a regression. I had no intention of breaking 2.0.0 support. I actually removed it from Travis because the build was broken for a different reason. D'oh.

I've fixed this in 187e207303d40ed8f5c6d6c50441ed0ad1d6b373, figured out the Travis issue and [released a new gem](https://rubygems.org/gems/slop/versions/4.4.3). Sorry for the trouble!

**Comment by bjoernalbers** [comment]
Awesome response time! Thanks 👍
